app/build.gradle.kts

```diff
@@ -35,6 +35,9 @@ android {
         versionName = "1.0"
 
         testInstrumentationRunner = "androidx.test.runner.AndroidJUnitRunner"
+
+        // this build config field is to set the maps api key in the manifest properly - from chatgpt :)
+        buildConfigField("String", "MAPS_KEY", "\"${project.findProperty("MAPS_KEY")}\"");
```

Comment:
I'm not sure if you need this anymore as the build config is generated from values `secrets.properties` but it's fine for now, can take a look into it in the future

Reply:
i tried taking this out my google maps still refused to load without this. maybe it's something wrong with my end but not sure what's causing it